src/probnum/filtsmooth/optim/_iterated_component.py

```diff
@@ -164,3 +164,22 @@ def __getattr__(self, attr):
             return self.attr
         else:
             return getattr(self._component, attr)
+
+    def _duplicate(self, **changes):
+        def replace_key(key):
+            """If the key is part of the desired changes, change appropriately.
+
+            Otherwise, take the current value.
+            """
+            try:
+                return changes[key]
+            except KeyError:
+                return getattr(self, key)
+
+        component = replace_key("component")
+        stopcrit = replace_key("stopcrit")
+
+        return IteratedDiscreteComponent(
+            component=component,
+            stopcrit=stopcrit,
+        )
```

Comment:
Some thoughts:
1) This function replicates the `dataclasse.replace` functionality (this is where the `**changes` notation stems from)
2) Maybe have a look at `markov.Transition.duplicate()` for more info
3) I am not suuuper happy with the implementation, but it does the job reliably.

Reply:
I left some comments in the other classes (below).